reacttodo/src/App.js

```diff
@@ -0,0 +1,93 @@
+import "./App.css";
+import React, { useState, useEffect } from "react";
+
+function App() {
+  const [data, setData] = useState([]);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  useEffect(() => {
+    const getTodoData = JSON.parse(localStorage.getItem("todoData")) ?? [];
+    setData(getTodoData);
+  }, []);
+
+  const submitHandler = () => {
+    if (!title || !description) return;
+    const currentData = [{ title, description }, ...data];
+    setData(currentData);
+    localStorage.setItem("todoData", JSON.stringify(currentData));
+    setDescription("");
+    setTitle("");
+  };
+
+  const deleteItem = (idx) => {
+    const updateData = data.filter((_, id) => idx !== id);
+    setData(updateData);
+    localStorage.setItem("todoData", JSON.stringify(updateData));
+  };
+  const updateItem = (todo, idx) => {
```

Comment:
@isagarjaiswal This code can be broken in the scenario.
On updating the item you are removing the item, right? what if the user does not complete the process example and does not submit the form? We will lose that data in this case. Can we implement the correct update functionality?  ( means just try to updated the targeted data instead of doing delete and add that data again )